src/Util/ProjectReorganiser.php

```diff
@@ -0,0 +1,249 @@
+        string $updatedProject = 'app',
+        string $legacyCode = 'code',
+        string $updatedCode = 'src'
+    ) {
+        $this->rootPath = $rootPath;
+        $this->legacyProject = $legacyProject;
+        $this->updatedProject = $updatedProject;
+        $this->legacyCode = $legacyCode;
+        $this->updatedCode = $updatedCode;
+    }
+
+    /**
+     * Check if there's a legacy project folder and if it can be moved.
+     * @return int
+     */
+    public function checkProjectFolder()
+    {
+        return $this->checkGenericFolder($this->legacyProject, $this->updatedProject);
+    }
+
+    /**
+     * Check if there's a legacy code folder and if it can be moved.
+     * @return int
+     */
+    public function checkCodeFolder()
+    {
+        $projectStatus = $this->checkProjectFolder();
+        if (in_array($projectStatus, [self::NOTHING, self::BLOCKED_LEGACY])) {
+            // If we have both `app` and `mysite` ... or neither, let's bail.
+            return $projectStatus;
+        }
+
+        // Pick whatever we'll be looking for our code folder in `mysite` or `app`.
+        $projectFolder = $projectStatus == self::ALREADY_UPGRADED ?
+            $this->updatedProject :
+            $this->legacyProject;
+
+        return $this->checkGenericFolder(
+            $projectFolder . DIRECTORY_SEPARATOR . $this->legacyCode,
+            $projectFolder . DIRECTORY_SEPARATOR . $this->updatedCode
+        );
+    }
+
+    /**
+     * Generic method for checking whatever a legacy and updated path are in use.
+     * @param  string $legacyPath
+     * @param  string $updatedPath
+     * @return int
+     */
+    private function checkGenericFolder($legacyPath, $updatedPath)
+    {
+        // Do we have a mysite? Is it a folder
+        $legacy = $this->folderExists($legacyPath) ?
+            self::UPGRADABLE_LEGACY :
+            self::NOTHING;
+
+        // Do we have an `app` folder? Doesn't matter if it's a folder really.
+        $updated = $this->exists($updatedPath) ?
+            self::ALREADY_UPGRADED :
+            self::NOTHING;
+
+        // Combine both possiblity.
+        return $legacy | $updated;
+    }
+
+
+    /**
+     * Try to rename legacy project folder.
+     * @param bool $dryrun Pretent to do the operation
+     * @throws LogicException
+     * @return array Files that are moved, the key being the original filename and the value being the destination
+     */
+    public function moveProjectFolder($dryrun = false): array
+    {
+        switch ($this->checkProjectFolder()) {
+            case self::BLOCKED_LEGACY:
+                throw new LogicException('Target folder already exists.');
+
+            case self::UPGRADABLE_LEGACY:
+                return $this->moveGenericFolder(
+                    $this->legacyProject,
+                    $this->updatedProject,
+                    $dryrun
+                );
+
+            case self::ALREADY_UPGRADED:
+            case self::NOTHING:
```

Comment:
Could just use `default` to prevent possible default case being missed.

Reply:
Done.